File: Algorithm by Swift/Algorithm by Swift/Stack/10773.swift
//
//  main.swift
//  Algorithm by Swift
//
//  Created by zaehorang on 2024/02/22.
//  https://www.acmicpc.net/problem/10773

var s = [Int]()

let n = Int(readLine()!)!

for _ in 1...n {
    let a = Int(readLine()!)!
    
    if a != 0 {
        s.append(a)
    } else {  // 0인 경우
//        if s.isEmpty { continue }  // 잘못된 숫자를 부를때마다 0을 외친다는 조건이 있으므로 비어있는 경우는 고려하지 않아도 될 듯.
        s.removeLast()
    }
}

print(s.reduce(0, +))

File: Algoritm by CPP/Algoritm by CPP/Stack/10773.cpp
//
//  main.cpp
//  Algoritm by CPP
//
//  Created by zaehorang on 2024/02/22.
//  https://www.acmicpc.net/problem/10773

#include <bits/stdc++.h>
using namespace std;

int main(void){
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    stack<int> s;
    
    int n;
    cin >> n;
    
    while(n--){
        int a;
        cin >> a;
        
        if(a == 0) s.pop();
        else s.push(a);
    }
    
    int t = 0;
    
    // 스택에 쌓여 있는 모든 값을 더해줌
    while(!s.empty()){
        t += s.top();
        s.pop();
    }
    cout << t;
}
